war/models.py

```diff
@@ -72,17 +72,26 @@ def bin_max(value):
         return value
 
 
-def survivor_signature(value):
-    machine_code = value.file.read()
-    if not machine_code:
-        raise ValidationError("Survivor may not be empty.")
+class SurvivorSignatureValidator:
```

Comment:
I think the different validators should be moved to `validators.py` so it's clearer. We do have to be mindful of circular imports though